In [28]:
import pandas as pd
import numpy as np

# **Extract data**

In [29]:
# function for read tất cả gg sheet
def read_ggsheet (google_sheet_id):
  url='https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
  df = pd.read_excel(url)
  return df
google_sheet_id = ['1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI',
                   '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c',
                   '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw',
                   '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4',
                   '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE']
for id in google_sheet_id:
  read_ggsheet(id)

In [30]:
customer_lifespan_data = pd.read_excel('https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/export?format=xlsx', sheet_name='Sheet1')

In [31]:
daily_marketing_spendings = pd.read_excel('https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/export?format=xlsx', sheet_name='Sheet1')

In [32]:
monthly_expenses = pd.read_excel('https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/export?format=xlsx', sheet_name='Sheet1')

In [33]:
payroll = pd.read_excel('https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/export?format=xlsx', sheet_name='Sheet1')

In [34]:
receipts_history = pd.read_excel('https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/export?format=xlsx', sheet_name='Sheet1')

In [35]:
def get_last_month(dataset,date_col):
  last_month = dataset[date_col].max().month
  cond = dataset[date_col].dt.month == last_month
  return dataset[cond]

In [36]:
customer_lifespan_data.head()


,Unnamed: 0,start_date,churn_date
0,1000,2021-11-15,2022-09-14
1,1001,2022-04-15,2023-02-16
2,1002,2022-10-30,2023-02-04
3,1003,2021-08-22,2023-02-07
4,1004,2021-08-23,2022-02-02


In [37]:
daily_marketing_spendings.head()

,date,channel,spending
0,2023-01-01,Google Ads,784
1,2023-01-01,Facebook Ads,659
2,2023-01-01,LinkedIn Ads,729
3,2023-01-01,Twitter Ads,292
4,2023-01-02,Google Ads,935


In [38]:
monthly_expenses.head()

,#,month,category,item,amount
0,1,2023-01-01,Server Costs,AWS Hosting,8000
1,2,2023-01-01,Server Costs,Google Cloud Storage,4000
2,3,2023-01-01,Software Licenses,Atlassian Jira,1200
3,4,2023-01-01,Software Licenses,Slack,800
4,5,2023-01-01,Software Licenses,Salesforce,1500


In [39]:
monthly_expenses.item.unique()

array(['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira', 'Slack',
       'Salesforce', 'Zoom', 'Office Rent', 'Office Supplies',
       'Travel Expenses'], dtype=object)

In [40]:
payroll.head()

,month,department,employee_name,position,paid
0,2023-01-01,Sales,John Doe,Sales Manager,1500
1,2023-01-01,Sales,Jane Smith,Sales Associate,600
2,2023-01-01,Sales,Jim Brown,Sales Associate,700
3,2023-01-01,Sales,Laura Miller,Sales Associate,800
4,2023-01-01,Marketing,Alice Johnson,Marketing Manager,1650


In [41]:
payroll.department.unique()

array(['Sales', 'Marketing', 'Engineering', 'HR', 'Support', 'Analytics'],
      dtype=object)

In [42]:
receipts_history.head()

,date,customer_id,receipt_amount,new_customer
0,2023-01-01,2653,67,1
1,2023-01-01,2731,271,1
2,2023-01-01,1277,231,0
3,2023-01-01,2094,107,0
4,2023-01-01,1314,416,0


In [43]:
daily_marketing_spendings = daily_marketing_spendings.convert_dtypes()

In [44]:
payroll = payroll.convert_dtypes()

In [45]:
monthly_expenses = monthly_expenses.convert_dtypes()

# **Calculation**

# **CAC - Customer Aquisition Cost**

In [46]:
# tính CRM

In [47]:
last_month = monthly_expenses.month.max()

In [48]:
last_month_expenses = monthly_expenses[monthly_expenses['month'] == last_month]

In [49]:
# cách 1
cond = last_month_expenses['item'] == 'Salesforce'
crm_cost = last_month_expenses[cond]['amount'].values[0]
print(crm_cost)

1700


In [50]:
# cách 2
CRM = monthly_expenses[(monthly_expenses['month'].dt.month == 3) & (monthly_expenses['item'] == 'Salesforce')]['amount'].sum()

print(CRM)

1700


In [51]:
# Marketing spending in mar 2023
Total_marketing_spending_mar = daily_marketing_spendings[daily_marketing_spendings['date'].dt.month == 3].spending.sum()
print(Total_marketing_spending_mar)

68830


In [80]:
# spending on each marketing chanel on mar
daily_marketing_spendings[(daily_marketing_spendings['date'].dt.month == 3)].groupby('channel')['spending'].sum()


,spending
channel,
Facebook Ads,18635
Google Ads,16738
LinkedIn Ads,15171
Twitter Ads,18286


In [89]:
# total sales (  sales and marketing department in mar)

df_sales_marketing = payroll[payroll['department'].isin(['Sales', 'Marketing'])]
df_sales_marketing.head()


,month,department,employee_name,position,paid
0,2023-01-01,Sales,John Doe,Sales Manager,1500
1,2023-01-01,Sales,Jane Smith,Sales Associate,600
2,2023-01-01,Sales,Jim Brown,Sales Associate,700
3,2023-01-01,Sales,Laura Miller,Sales Associate,800
4,2023-01-01,Marketing,Alice Johnson,Marketing Manager,1650


In [54]:
total_sales_mar = df_sales_marketing[df_sales_marketing['month'].dt.month == 3].paid.sum()
print(total_sales_mar)

5950


In [86]:
df_sales_marketing[df_sales_marketing['month'].dt.month == 3].count()
# there are 6 staffs in sales and marketing department

,0
month,6
department,6
employee_name,6
position,6
paid,6


In [55]:
# total new customer in mar 2023
total_newcus_mar = receipts_history[receipts_history['date'].dt.month == 3].new_customer.sum()
print(total_newcus_mar)

63


In [56]:
# CAC
CAC = (CRM + Total_marketing_spending_mar + total_sales_mar )/total_newcus_mar
print(CAC)

1213.968253968254


# **ARPU - Average revenue per user**

In [57]:
# Total number of user in Mar
total_user_mar = receipts_history[receipts_history['date'].dt.month == 3].customer_id.count()
print(total_user_mar)

311


In [58]:
# total revenue in mar
total_revenue_mar = receipts_history[receipts_history['date'].dt.month == 3].receipt_amount.sum()
print(total_revenue_mar)

83033


In [59]:
# ARPU
ARPU = total_revenue_mar / total_user_mar
print(ARPU)

 # meaning each customers generated $266.98 in revenue on average in mar

266.9871382636656


# **COGS - Cost of Goods Sold**

In [60]:
# PRODUCTION COST  = AWS Hosting,  Google Cloud Storage, Atlassian Jira

In [61]:
production_expense_tab = monthly_expenses[monthly_expenses['item'].isin(['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira'])]

In [62]:
production_cost = production_expense_tab[
    production_expense_tab['month'].dt.month == 3
]['amount'].sum()

In [63]:
production_cost

np.int64(14200)

In [95]:
# epense for each item (production cost) in mar
production_expense_tab[production_expense_tab['month'].dt.month == 3].groupby('item')['amount'].sum()

,amount
item,
AWS Hosting,8400
Atlassian Jira,1400
Google Cloud Storage,4400


In [64]:
from typing import SupportsFloat
# solfware cost : item slack and zoom
solfware_table = monthly_expenses[monthly_expenses['item'].isin(['Slack', 'Zoom'])]
solfware_cost = solfware_table[solfware_table['month'].dt.month == 3]['amount'].sum() * 0.6
solfware_cost

np.float64(864.0)

In [65]:
# production_salary_cost = department - eninerring

production_salary_cost = payroll[(payroll['month'].dt.month == 3) & (payroll['department'] == 'Engineering')]['paid'].sum()
production_salary_cost

np.int64(5200)

In [96]:
payroll[(payroll['month'].dt.month == 3) & (payroll['department'] == 'Engineering')].count()
# there are 3 staff in Engineering deparment

,0
month,3
department,3
employee_name,3
position,3
paid,3


In [66]:
# COGS

COGS = production_cost + solfware_cost + production_salary_cost
print(COGS)


20264.0


# **Gross Margin**

In [67]:
# revenue
revenue = receipts_history[receipts_history['date'].dt.month == 3].receipt_amount.sum()
print(revenue)

83033


In [68]:
gross_margin = (revenue - COGS)/revenue * 100
print(gross_margin)

75.5952452639312


# **LTV - lifetime value**

In [101]:
customer_lifespan = (customer_lifespan_data.churn_date - customer_lifespan_data.start_date).dt.days


In [70]:
average_customer_lifespan = customer_lifespan.mean()
average_customer_lifespan_permonth = average_customer_lifespan/30
print(average_customer_lifespan_permonth)

9.841333333333333


In [71]:
LTV = ARPU * average_customer_lifespan_permonth * gross_margin/100
print(LTV)

1986.272192926045


# **LTV / CAC**

In [72]:
LTV/CAC

np.float64(1.6361813304699375)

# **Conclusion**

- CAC = 1213.97 pretty high
- ARPU = 266.99 the Average Revenue Per User
(ARPU)  is 266.99, meaning each customers generated $266.99 in revenue on average
during the month.
- COGS = 20264 costs of producing the goods is 20264 dola
- Gross margin = 75.59%   gross margin is 75.59% which higher than 50% is
considered as a pretty good indicator. It is scalable
- LTV = 1986.27 Lifetime value of a customer is 1986.27 dola
- LTV/CAC = 1.636 A ratio greater than 1 indicates that the company is generating more revenue from
customers than it costs to acquire them. This is a positive sign, suggesting profitability
and sustainable growth. However, this is not ideal ratio


# **Recommendations**

- CAC is pretty high, especially spending on marketings which is quite high. We should evaluate the effectiveness of each marketing channel and consider reduce cost for the less effective channel. Moreover, also salary of markerting manager is highest, however the efficiency of marketing need to be re-evaluate since we spent alot on it. After evaluating, we can reconsider about salary which we paid with their contributions.
- Gross margin is 75.59% this number show that we can scale Streamline Pro
- opimize COGS : production cost is majority, and AWS Hosting is double the rest. We can consider reduce this cost by finding alternative provider, ask advice from engineering to optimize cost for AWS Hosting.
- Increse LTV : Reduce COGS as suggestion above, and consider clasify customer type to support customer better, from that can increase churn